### ISSUE TITLE
Issue with Pull Environments

### ISSUE THREAD

**Issue by haarchri** [mvisonneau/gitlab-ci-pipelines-exporter#191]
Hello,
i have a Problem with Pull Environments

```
panic: runtime error: invalid memory address or nil pointer dereference
--
  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0xad4528]
  |  
  | goroutine 2199 [running]:
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/gitlab.(*Client).GetEnvironment(0xc0001cafc0, 0xc000533800, 0x1c, 0x2b8b1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/gitlab/environments.go:76 +0x288
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter.updateEnvironment(0xc0003cb658, 0xc000578820, 0x9)
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter/environments.go:52 +0x65
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter.pullEnvironmentsFromProject(0xc0001236fb, 0xc0001eef70, 0x0, 0xc0001eefe0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter/environments.go:35 +0x2c5
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter.glob..func2(0xc0001236fb, 0xc0001eef70, 0x0, 0xc0001eefe0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  | github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter/scheduler.go:24 +0x58
  | reflect.Value.call(0xbe9b20, 0xd6c058, 0x13, 0xd38030, 0x4, 0xc0004ff720, 0x1, 0x1, 0xc0003cbda8, 0x47774f, ...)
  | reflect/value.go:476 +0x8c7
  | reflect.Value.Call(0xbe9b20, 0xd6c058, 0x13, 0xc0004ff720, 0x1, 0x1, 0x0, 0xc0003cbde8, 0xc0003cbdf8)
  | reflect/value.go:337 +0xb9
  | github.com/vmihailenco/taskq/v3.(*reflectFunc).HandleMessage(0xc0001ca2a0, 0xc00038d450, 0xd4df69, 0x1f)
  | github.com/vmihailenco/taskq/v3@v3.2.1/handler.go:78 +0x8d
  | github.com/vmihailenco/taskq/v3.(*Task).HandleMessage(0xc0001ca2d0, 0xc00038d450, 0x1f, 0xc0001ca2d0)
  | github.com/vmihailenco/taskq/v3@v3.2.1/task.go:112 +0xa2
  | github.com/vmihailenco/taskq/v3.(*TaskMap).HandleMessage(0x1297600, 0xc00038d450, 0x0, 0x0)
  | github.com/vmihailenco/taskq/v3@v3.2.1/registry.go:71 +0xb3
  | github.com/vmihailenco/taskq/v3.(*Consumer).Process(0xc0002903c0, 0xc00038d450, 0xc000122418, 0xc0000ca140)
  | github.com/vmihailenco/taskq/v3@v3.2.1/consumer.go:538 +0x14d
  | github.com/vmihailenco/taskq/v3.(*Consumer).worker(0xc0002903c0, 0xe1d260, 0xc000122418, 0xc000000001)
  | github.com/vmihailenco/taskq/v3@v3.2.1/consumer.go:479 +0x176
  | github.com/vmihailenco/taskq/v3.(*Consumer).addWorker.func1(0xc0002903c0, 0xe1d260, 0xc000122418, 0xc000000001)
  | github.com/vmihailenco/taskq/v3@v3.2.1/consumer.go:275 +0x74
  | created by github.com/vmihailenco/taskq/v3.(*Consumer).addWorker
  | github.com/vmihailenco/taskq/v3@v3.2.1/consumer.go:273 +0xd3
```

config looks like this:

```
...
    project_defaults:
      pull:
        pipeline:
          jobs:
            enabled: true
          environments:
            enabled: true
    wildcards:
      - owner:
          name: abc
          kind: group
        pull:
          refs:
            regexp: '^(master|release|development)$'
      - owner:
          name: def
          kind: group
        pull:
          refs:
            regexp: '^(master|release|feature|develop|development)$'
...
```

any ideas what is the problem ? with environments enabled false the exporter works like a charme

**Comment by mvisonneau** [comment]
👋 hey @haarchri, interesting issue, thanks for raising it! Which version of the exporter are you using?

At first sight, it doesn't explain the error but the configuration syntax should be:

```yaml
project_defaults:
  pull:
    environments:
      enabled: true

    pipeline:
      jobs:
        enabled: true

wildcards:
  - owner:
      name: abc
      kind: group
    pull:
      refs:
        regexp: '^(master|release|development)$'
  - owner:
      name: def
      kind: group
    pull:
      refs:
        regexp: '^(master|release|feature|develop|development)$'
```

I will look into the issue though!

**Comment by mvisonneau** [comment]
Alright, should be fixed as part of v0.4.5 @haarchri. Feel free to reopen if not!
